modules/cloud-provider/default.nix

```diff
@@ -0,0 +1,113 @@
+{ lib, config, modulesPath, disko, ... }:
+
+with lib;
+
+let
+
+  # Recursively constructs an attrset of a given folder, recursing on directories, value of attrs is the filetype
+  getDir = dir:
+    mapAttrs
+      (file: type: if type == "directory" then getDir "${dir}/${file}" else type)
+      (builtins.readDir dir);
+
+  # Collects all files of a directory as a list of strings of paths
+  files = dir:
+    collect isString
+      (mapAttrsRecursive (path: type: concatStringsSep "/" path) (getDir dir));
+
+  # Filters out directories that don't end with .nix or are this file, also makes the strings absolute
+  validFiles = dir:
+    map (file: ./. + "/${file}") (filter (file: hasSuffix ".nix" file && file != "default.nix") (files dir));
+
+  cfg = config.lounge-rocks.cloud-provider;
+
+in
+{
+  imports = validFiles ./. ++ [
+    disko.nixosModules.disko
+    (modulesPath + "/installer/scan/not-detected.nix")
+    (modulesPath + "/profiles/qemu-guest.nix")
+  ];
+
+  options.lounge-rocks.cloud-provider = {
+    enable = mkEnableOption "cloud-provider";
+    primaryDisk = mkOption {
+      type = types.str;
+      default = "/dev/sda";
+      description = "The name of the primary disk";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    # Running fstrim weekly is a good idea for VMs.
+    # Empty blocks are returned to the host, which can then be used for other VMs.
+    # It also reduces the size of the qcow2 image, which is good for backups.
+    services.fstrim = {
+      enable = true;
+      interval = "weekly";
+    };
+
+    # We want to standardize our partitions and bootloaders across all providers.
+    # See: https://github.com/lounge-rocks/the-lounge/issues/16
+    # Currently Hetzner and Proxmox are standardized.
+    disko.devices.disk.main = mkIf (cfg.hetzner.enable || cfg.proxmox.enable) {
```

Comment:
If this is dependent on `hetzner.enable` or `proxmox.enable` why not just put it in their respective options?

Reply:
TLDR: our Netcup & Oracle servers need to be redeployed - after that, the check won't be necessary.
Every provider will share the same bootloader and partition config.
There isn't that much provider specific stuff after that (mainly kernel modules & boot options for the serial console).